src/ReaderImpl.cpp

```diff
@@ -0,0 +1,1276 @@
+/*
+ * Copyright (c) 2010 Stan Coleby (scoleby@intelisum.com)
+ * Copyright (c) 2020 PTC Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person or organization
+ * obtaining a copy of the software and accompanying documentation covered by
+ * this license (the "Software") to use, reproduce, display, distribute,
+ * execute, and transmit the Software, and to prepare derivative works of the
+ * Software, and to permit third-parties to whom the Software is furnished to
+ * do so, all subject to the following:
+ *
+ * The copyright notices in the Software and this entire statement, including
+ * the above license grant, this restriction and the following disclaimer,
+ * must be included in all copies of the Software, in whole or in part, and
+ * all derivative works of the Software, unless such copies or derivative
+ * works are solely in the form of machine-executable object code generated by
+ * a source language processor.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE, TITLE AND NON-INFRINGEMENT. IN NO EVENT
+ * SHALL THE COPYRIGHT HOLDERS OR ANYONE DISTRIBUTING THE SOFTWARE BE LIABLE
+ * FOR ANY DAMAGES OR OTHER LIABILITY, WHETHER IN CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+#include "ReaderImpl.h"
+
+namespace e57
+{
+
+   ReaderImpl::ReaderImpl( const ustring &filePath ) :
+      imf_( filePath, "r" ), root_( imf_.root() ), data3D_( root_.get( "/data3D" ) ),
+      images2D_( root_.get( "/images2D" ) )
+   {
+   }
+
+   ReaderImpl::~ReaderImpl()
+   {
+      if ( IsOpen() )
+         Close();
```

Comment:
Please add braces {} around ifs.

I am going to do this for the whole code base in an upcoming commit. If it's not easy for you to do this, I can use clang-tidy to do it.

Reply:
Done with clang-tidy.